### PR TITLE
[compute_tools] Add /insights endpoint to compute_ctl

### PR DESCRIPTION
**Pull request by ololobus**
## Describe your changes

This commit adds a basic HTTP API endpoint that allows scraping the `pg_stat_statements` data and getting a list of slow queries. New insights like cache hit rate and so on could be added later.

Extension `pg_stat_statements` is checked / created only if compute tries to load the corresponding shared library. The latter is configured by control-plane and currently covered with feature flag.

Co-authored by Eduard Dyckman (bird.duskpoet@gmail.com)

## Checklist before requesting a review
- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

### CODE REVIEW

File: compute_tools/src/pg_helpers.rs [hunk 1]
```diff
@@ -63,12 +63,7 @@ impl GenericOption {
     /// Represent `GenericOption` as configuration option.
     pub fn to_pg_setting(&self) -> String {
         if let Some(val) = &self.value {
-            let name = match self.name.as_str() {
-                "safekeepers" => "neon.safekeepers",
-                "wal_acceptor_reconnect" => "neon.safekeeper_reconnect_timeout",
-                "wal_acceptor_connection_timeout" => "neon.safekeeper_connection_timeout",
-                it => it,
-            };
```
Comment (lubennikovaav) [inline compute_tools/src/pg_helpers.rs:71]: This is unrelated to this PR, right?
Do we know for sure that all tenants have these updated?

**Review by lubennikovaav — APPROVED** [review]

File: compute_tools/src/pg_helpers.rs (hunk 1 above)
Comment (ololobus) [reply compute_tools/src/pg_helpers.rs:71]: I've checked that in the console code we don't set them for quite a while, but this is a good point that we probably should check console db too. Will put a TODO, double check and remove later in a separate PR

File: compute_tools/src/compute.rs [hunk 2]
```diff
@@ -50,6 +51,19 @@ pub struct ComputeNode {
...
+pub struct InsightsRow {
+    pub query: String,
+    pub total_runtime: String,
+    pub mean_runtime: String,
+}
+
+#[derive(Serialize)]
+pub struct Insights {
+    count: u64,
+    statements: Vec<InsightsRow>,
```
Comment (vadim2404) [inline compute_tools/src/compute.rs:64]: I didn't see where you used these structs.

Comment (ololobus) [reply compute_tools/src/compute.rs]: Good catch, thanks. Some leftover from the previous version of the patch

File: compute_tools/src/compute.rs [hunk 3]
```diff
@@ -400,4 +402,39 @@ impl ComputeNode {
...
+FROM
+    pg_stat_statements
+WHERE
+    userid != 'cloud_admin'::regrole::oid
+ORDER BY
+    (mean_exec_time + mean_plan_time) DESC
+LIMIT 100",
+            )
+            .await)
+            .unwrap()
```
Comment (vadim2404) [inline compute_tools/src/compute.rs:430]: Potentially, it can panic here. Make it safer a bit.

I assume that here it is less dangerous, but still.

```rust
row.get(0).unwrap().to_string()
```

Comment (ololobus) [reply compute_tools/src/compute.rs:430]: > I assume that here it is less dangerous

Yes, if it panics, then request handler just panics and nothing special will happen (I suppose)

Is your code snippet a suggestion? Then I'm not sure where to put it?

Comment (vadim2404) [reply compute_tools/src/compute.rs:430]: nope, there are two places with `unwrap`

Comment (ololobus) [reply compute_tools/src/compute.rs]: I got rid of unnecessary unwraps

**Review by vadim2404 — APPROVED** [review]
